### DISCUSSION TITLE
Add support for controller gyro's and accelerometer's

### DISCUSSION THREAD

**Discussion by RetrocadeMedia** [godotengine/godot-proposals#4420] [General Discussions]
Currently Godot does not support the gyro or accelerometers inside some controllers like the PS4 Dualshock or the Switch Pro Controller. Godot is clearly capable of utilizing them, as we can easily access the accelerometer of a phone, and it has support for VR (although admittedly I don't entirely understand the implementation of VR support, so for all I know Godot is just given position and rotation values, but my assumption was that Godot has to handle some of the tracking).

I understand that supporting these accelerometers isn't a huge priority as far fewer people are making games with motion controls these days, but motion control support would be useful for Switch developers, as well as for gyro aiming in FPS games which while controversial can provide more accuracy or even be a useful accessibility feature. It can also just be a fun thing to tinker with, I personally just want to add some stupid shake the controller type controls to a prototype I'm working on.

If it doesn't make sense to add directly to Godot, is this the sort of thing that could be done through a plugin?

**Comment by Zireael07** [comment]
My understanding is that Godot uses an external database of controller data. So it's not somethng easily fixed in Godot itself.

**Comment by Calinou** [comment]
This is already being tracked in https://github.com/godotengine/godot-proposals/issues/2829.

**Reply by RetrocadeMedia** [reply]
Ah, that it is! I hope there's some kind of update in there soon
